packages/dev/docs/pages/blog/creating-a-pointer-friendly-submenu-experience.mdx

```diff
@@ -0,0 +1,108 @@
+
+```tsx snippet
+import {SubmenuAnimation} from './SubmenuAnimation';
+<SubmenuAnimation />
+```
+
+## Predicting User Intent
+
+We can predict the user’s intent by tracking: 
+
+* Pointer movement **direction**
+* Pointer movement **speed**
+
+We can do this by listening for [pointermove](https://developer.mozilla.org/en-US/docs/Web/API/Element/pointermove_event) events and analyzing the changes in the pointer’s position (also called delta).
+
+## Valid Movements
+
+Now let’s imagine two lines: one from the pointer to the top of the submenu, and one from the pointer to the bottom of the submenu. We now essentially have a “range” of where the user might move their pointer on its path to the submenu. Any movement outside of this range should be considered invalid, and the submenu should close if the pointer moves outside of the original menu option.
+
+<SubmenuSafeArea
+  role="img"
+  aria-label="Diagram showing safe area triangle drawn between pointer and submenu" />
+
+## Arctangent
+
+We can measure the angle at which the pointer moved by using the 2-argument arctangent, or [atan2](https://en.wikipedia.org/wiki/Atan2). If we provide our pointer’s delta x and delta y values as arguments, we’ll get the angle (in radians) from the previous pointer position to the current pointer position.
+
+<SubmenuAtan2
+  role="img"
+  aria-label="Diagram showing atan2 value as angle between the X axis and some line on a cartesian plane" />
+
+Now we can use the atan2 function to measure the angles formed by three separate lines:
+
+* **Θ<sub>top</sub>** : Angle formed by the line from the previous pointer position to the **top inside corner** of the submenu
+* **Θ<sub>bottom</sub>** : Angle formed by the line from the previous pointer position to the **bottom inside corner** of the submenu
+* **Θ<sub>pointer</sub>** : Angle formed by the line from the previous pointer position to the **current pointer** position (delta)
+
+<SubmenuAtan2FromPointer
+  role="img"
+  aria-label="Diagram showing triangles formed between pointer and various points on submenu" />
+
+If the pointer’s delta angle is **between** the top and bottom angles, we know the user is moving their pointer in the direction of the submenu.
+
+<p style={{textAlign: 'center', fontSize: 'xx-large'}}>Θ<sub>top</sub> > Θ<sub>pointer</sub> > Θ<sub>bottom</sub></p>
+
+## Time
+
+We also want to consider the pointer’s movement over time when predicting the user’s intent. A user typically increases the **speed** of their pointer when moving towards the submenu, then decreases it as they reach their target. The user might change their mind on the way, however, and slow down to a stop or keep browsing other options in the parent menu. If their pointer speed dips below a certain threshold before arriving at the submenu, we could assume they're no longer intending to go to the submenu.
```

Comment:
```suggestion
In order to predict the user’s intent, we also need to know the pointer's velocity. A user typically accelerates their pointer when moving towards the submenu, then decelerates as they reach their target.

The user can slow down to a stop or continue browsing other options in the parent menu. If their pointer speed dips below a certain threshold before arriving at the submenu, we could assume they're no longer intending to go there or the submenu was opened accidentally.
```

Reply:
I'm still not quite happy with my suggestion, it's hard to tell how this relates to speed as opposed to angle: "continue browsing other options in the parent menu"

Reply:
I was more talking about stopping your pointer movement to read the options in the submenu (after you've already left the trigger item). I can definitely reword that.